controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -203,6 +206,30 @@ func add(mgr manager.Manager, r reconcile.Reconciler, ci hcoutil.ClusterInfo) er
 		}
 	}
 
+	apiServerCRPlaceholder, err := getApiServerCRPlaceholder()
+	if err != nil {
+		return err
+	}
+
+	if ci.IsOpenshift() {
+		// Watch openshiftconfigv1.APIServer separately
+		msg := "Reconciling for openshiftconfigv1.APIServer"
+		err = c.Watch(
+			&source.Kind{Type: &openshiftconfigv1.APIServer{}},
+			handler.EnqueueRequestsFromMapFunc(func(a client.Object) []reconcile.Request {
+				// enqueue using a placeholder to signal that the change is not
+				// directly on HCO CR but on the APIServer CR that we want to reload
+				// only if really changed
+				log.Info(msg)
+				return []reconcile.Request{
+					{NamespacedName: apiServerCRPlaceholder},
+				}
+			}),
+		)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Please consider to take it out to a new function, as sonarCloud warn about a code smell here (too complex function).